pkg/operator/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -200,53 +229,75 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 	return false, nil
 }
 
-func manageOpenShiftAPIServerEtcdCerts_v311_00_to_latest(client coreclientv1.CoreV1Interface, recorder events.Recorder) (bool, error) {
+func manageOpenShiftAPIServerEtcdCerts_v311_00_to_latest(client coreclientv1.CoreV1Interface, recorder events.Recorder) ([]runtime.Object, bool, error) {
 	const etcdServingCAName = "etcd-serving-ca"
 	const etcdClientCertKeyPairName = "etcd-client"
 
-	_, caChanged, err := resourceapply.SyncConfigMap(client, recorder, etcdNamespaceName, etcdServingCAName, targetNamespaceName, etcdServingCAName)
+	var modifiedObjects []runtime.Object
+
+	configMap, caChanged, err := resourceapply.SyncConfigMap(client, recorder, etcdNamespaceName, etcdServingCAName, targetNamespaceName, etcdServingCAName)
 	if err != nil {
-		return false, err
+		return nil, false, err
 	}
-	_, certKeyPairChanged, err := resourceapply.SyncSecret(client, recorder, etcdNamespaceName, etcdClientCertKeyPairName, targetNamespaceName, etcdClientCertKeyPairName)
+	if caChanged {
+		modifiedObjects = append(modifiedObjects, configMap)
+	}
+
+	secret, certKeyPairChanged, err := resourceapply.SyncSecret(client, recorder, etcdNamespaceName, etcdClientCertKeyPairName, targetNamespaceName, etcdClientCertKeyPairName)
 	if err != nil {
-		return false, err
+		return modifiedObjects, false, err
 	}
-	return caChanged || certKeyPairChanged, nil
+	if certKeyPairChanged {
+		modifiedObjects = append(modifiedObjects, secret)
+	}
+
+	return modifiedObjects, caChanged || certKeyPairChanged, nil
 }
 
-func manageOpenShiftAPIServerClientCA_v311_00_to_latest(client coreclientv1.CoreV1Interface, recorder events.Recorder) (bool, error) {
+func manageOpenShiftAPIServerClientCA_v311_00_to_latest(client coreclientv1.CoreV1Interface, recorder events.Recorder) (*corev1.ConfigMap, bool, error) {
 	const apiserverClientCA = "client-ca"
-	_, caChanged, err := resourceapply.SyncConfigMap(client, recorder, kubeAPIServerNamespaceName, apiserverClientCA, targetNamespaceName, apiserverClientCA)
-	if err != nil {
-		return false, err
+	configMap, caChanged, err := resourceapply.SyncConfigMap(client, recorder, kubeAPIServerNamespaceName, apiserverClientCA, targetNamespaceName, apiserverClientCA)
+	switch {
+	case err != nil:
+		return nil, false, err
+	case caChanged:
+		return configMap, true, nil
+	default:
+		return nil, false, nil
 	}
-	return caChanged, nil
 }
 
-func manageOpenShiftAPIServerImageImportCA_v311_00_to_latest(openshiftConfigClient openshiftconfigclientv1.ConfigV1Interface, client coreclientv1.CoreV1Interface, recorder events.Recorder) (bool, error) {
+func manageOpenShiftAPIServerImageImportCA_v311_00_to_latest(openshiftConfigClient openshiftconfigclientv1.ConfigV1Interface, client coreclientv1.CoreV1Interface, recorder events.Recorder) (*corev1.ConfigMap, bool, error) {
```

Comment:
@sanchezl start a pull on top of something like https://github.com/openshift/cluster-kube-apiserver-operator/pull/185 that moves this into configobservation

Reply:
Should I wait for https://github.com/openshift/library-go/pull/181 ?